GitUI/Script/ScriptRunner.cs

```diff
@@ -115,6 +116,11 @@ private static bool RunScript(IWin32Window owner, GitModule module, ScriptInfo s
 
             if (command.StartsWith(NavigateToPrefix))
             {
+                if (revisionGrid == null)
+                {
+                    return false;
+                }
```

Comment:
Should this check be significantly higher?
`revisionGrid` is used in other places in the method in a way that would result in a NRE... E.g. line:109

Reply:
Good point. I was able to produce a NRE for a `plugin:` script. However, in the future there might be other types of scripts which would not require the `revisionGrid` instance, so I'd rather not abort running the script if it is `null` in the general case. Instead, I'd just do the null check for the `plugin:` scripts the same way I did for the `navigateTo:` scripts. Is that OK?

Reply:
Hmm, after giving it a second thought, should we really not allow a script to run without the revision grid? It might not actually need it. We should just make sure it doesn't crash when creating the `GitUIEventArgs` instance (at line 109).

Reply:
> should we really not allow a script to run without the revision grid?

No, we should not. Scripts must be able to run without the revision grid - #6135 is a perfect example of the use case.

Reply:
For the `plugin:` scripts, the revision grid is only required to get a `GitUICommands` instance (through its `UICommands` property). Is it possible to get it in a different way? Or should we just pass `null` instead into the `GitUIEventArgs` constructor?

Reply:
`NavigateTo` scripts must only be available in the revision grid context, since it is a command performed on the grid itself.
We should be able to supply the `UICommands` instance to all scripts since it has a number of useful events and properties. 
I will need to investigate when I have time. Feel free to offer your vision.